service/task.go

```diff
@@ -524,8 +525,16 @@ func (uis *UIServer) taskLog(w http.ResponseWriter, r *http.Request) {
 		var logReader io.ReadCloser
 		logReader, err = uis.getBuildloggerLogs(r.Context(), projCtx, r, logType, DefaultLogMessages, execution)
 		if err == nil {
-			gimlet.WriteText(w, logReader)
-			grip.Warning(logReader.Close())
+			defer func() {
+				err := logReader.Close()
+				if err != nil {
```

Comment:
you don't need to if-err because the conditional logging, so this can all be in one statment